### PR TITLE
Cargo boxcutter

### PR DESCRIPTION
**Pull request by Shadyyy66**
## About The Pull Request

This PR adds a very simple boxcutter to the game, it does as much damage as an unsharpened kitchen knife, makes a fun little noise when being unfolded and folded and can very quickly unwrap wrapped packages and return the wrapping paper that was used to wrap without making a mess that needs to be cleaned up, it even has a custom sound for unwrapping!

This PR also adds a new proc for how parcels or wrapped objects in sortingmachinery.dm are opened by adding the "tear_open" proc, if it's set to TRUE, it takes time to unwrap and it leaves a mess, but if it's set to FALSE, it is instantly unwrapped and drops one (1) package wrapping and plays the custom box cutting sound clip I added.

## Why It's Good For The Game

Honestly, I just think it would be fun for cargo to have a tool of their own that isn't just a scanner or whatever they can make with the autolathe. Plus, as a frequent janitor player, it leaves less messes in cargo that janitors usually don't have access to anyway, and wastes less package wrapping!

Not only that, but it feels like cargo itself can be lacking in department-specific weapons to use. Every other department has some kind of weapon tied to it, like service and its knives and hatchets, or medical with its scalpels and saws.
Basically, it feels like cargo isn't living up to the same potential as the other departments, and mining might as well be its own department with how separate they are.

And honestly, cargo techs deserve some kind of weapon to defend themselves with, feels like there's always something going down in cargo.

## Changelog

:cl:
add: Added boxcutter to cargo.
/:cl:

Untitled-3

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
image

File: code/game/objects/items/boxcutter.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	force = 0
+	var/start_extended = FALSE
+	var/on_sound = 'sound/items/boxcutter_activate.ogg'
+
+/obj/item/boxcutter/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob, ITEM_SLOT_HANDS)
+	AddComponent(/datum/component/butchering, \
+	speed = 7 SECONDS, \
+	effectiveness = 100, \
```
Comment (Mothblocks) [inline code/game/objects/items/boxcutter.dm:23]: Indent up by one

File: code/game/objects/items/boxcutter.dm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	AddComponent(/datum/component/transforming, \
+		start_transformed = start_extended, \
+		force_on = 8, \
+		throwforce_on = 3, \
+		throw_speed_on = throw_speed, \
+		sharpness_on = SHARP_EDGED, \
+		hitsound_on = 'sound/weapons/bladeslice.ogg', \
+		w_class_on = WEIGHT_CLASS_NORMAL, \
+		attack_verb_continuous_on = list("cuts", "stabs", "slashes"), \
+		attack_verb_simple_on = list("cut", "stab", "slash"))
```
Comment (Mothblocks) [inline code/game/objects/items/boxcutter.dm:34]:
```suggestion
		attack_verb_simple_on = list("cut", "stab", "slash"), \
	)
```

File: code/game/objects/items/boxcutter.dm [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+	RegisterSignal(src, COMSIG_TRANSFORMING_ON_TRANSFORM, .proc/on_transform)
+
+/obj/item/boxcutter/proc/on_transform()
+	SIGNAL_HANDLER
+	playsound(src, on_sound, 50)
+	return COMPONENT_NO_DEFAULT_MESSAGE
```
Comment (Mothblocks) [inline code/game/objects/items/boxcutter.dm:41]: How noticable is the in-hand sprite being changed? Can you show a picture with switch blade extended and not? The message should only be silenced if looking at the center of your screen makes it obvious that you've activated it--we don't want people to have to move their eyes from where they're looking/rely on the sound alone.

File: code/game/objects/items/boxcutter.dm [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		hitsound_on = 'sound/weapons/bladeslice.ogg', \
+		w_class_on = WEIGHT_CLASS_NORMAL, \
+		attack_verb_continuous_on = list("cuts", "stabs", "slashes"), \
+		attack_verb_simple_on = list("cut", "stab", "slash"))
+
+	RegisterSignal(src, COMSIG_TRANSFORMING_ON_TRANSFORM, .proc/on_transform)
+
+/obj/item/boxcutter/proc/on_transform()
+	SIGNAL_HANDLER
+	playsound(src, on_sound, 50)
```
Comment (JohnFulpWillard) [inline code/game/objects/items/boxcutter.dm:40]: You can instead make this part of the transforming component, since it's currently hardcoded to batonextend.

File: code/modules/recycling/sortingmachinery.dm [hunk 5]
```diff
@@ -185,6 +188,15 @@
 			wrapped_item.AddComponent(/datum/component/pricetag, sticker.payments_acc, sticker.cut_multiplier)
 		update_appearance()
 
+	if(istype(item, /obj/item/boxcutter))
+		if( (item.icon_state == "boxcutter_on"))
```
Comment (JohnFulpWillard) [inline code/modules/recycling/sortingmachinery.dm:192]: The boxcutter should have an 'on' var instead, looking at the icon state is a bad idea,

File: code/modules/recycling/sortingmachinery.dm [hunk 6]
```diff
@@ -185,6 +188,15 @@
 			wrapped_item.AddComponent(/datum/component/pricetag, sticker.payments_acc, sticker.cut_multiplier)
 		update_appearance()
 
+	if(istype(item, /obj/item/boxcutter))
```
Comment (Mothblocks) [inline code/modules/recycling/sortingmachinery.dm:191]:
```suggestion
	else if(istype(item, /obj/item/boxcutter))
```

Otherwise `..()` will hit for all the other items.

Also I kind of hate that this entire proc looks like this but probably not worth it for you to change right now.

File: code/modules/recycling/sortingmachinery.dm (hunk 5 above)
Comment (Mothblocks) [inline code/modules/recycling/sortingmachinery.dm:192]: Have a var that updates instead of checking icon_state directly.

File: code/modules/recycling/sortingmachinery.dm [hunk 7]
```diff
@@ -185,6 +188,15 @@
...
+		if( (item.icon_state == "boxcutter_on"))
+			unwrap_contents(user)
+			post_unwrap_contents(user, rip_open = FALSE)
+			return
+		else
+			balloon_alert(user, span_warning("The boxcutter isn't active!"))
```
Comment (JohnFulpWillard) [inline code/modules/recycling/sortingmachinery.dm:197]:
```suggestion
			balloon_alert(user, span_warning("open [item] first!"))
```

File: code/modules/recycling/sortingmachinery.dm [hunk 8]
```diff
@@ -185,6 +188,15 @@
 			wrapped_item.AddComponent(/datum/component/pricetag, sticker.payments_acc, sticker.cut_multiplier)
 		update_appearance()
 
+	if(istype(item, /obj/item/boxcutter))
+		if( (item.icon_state == "boxcutter_on"))
+			unwrap_contents(user)
```
Comment (JohnFulpWillard) [inline code/modules/recycling/sortingmachinery.dm:193]: can you make this use the full unwrap code instead of shortcut to the end? It still takes time to open packages with a boxcutter

File: code/modules/recycling/sortingmachinery.dm (hunk 7 above)
Comment (Mothblocks) [inline code/modules/recycling/sortingmachinery.dm:197]:
```suggestion
			balloon_alert(user, "prime the boxcutter!")
```

Balloon alerts are not styled the same way as chat messages, and the wording on this wasn't very boxcutter-y. Feel free to replace prime with arm or ready.

File: code/modules/recycling/sortingmachinery.dm [hunk 9]
```diff
@@ -185,6 +188,15 @@
...
+			post_unwrap_contents(user, rip_open = FALSE)
+			return
```
Comment (Mothblocks) [inline code/modules/recycling/sortingmachinery.dm:195]: Remove both these since they always return anyway.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/recycling/sortingmachinery.dm (hunk 7 above)
Comment (Mothblocks) [reply code/modules/recycling/sortingmachinery.dm:197]: this is wrong because it's a balloon alert and balloon alerts should not have spans and should be kept as small as possible

**Comment by optimumtact** [comment]
maintainers add this

File: code/modules/recycling/sortingmachinery.dm (hunk 7 above)
Comment (JohnFulpWillard) [reply code/modules/recycling/sortingmachinery.dm:197]: problem is 'open it first' or something makes you question if it means the boxcutter or the wrapper

Comment (Mothblocks) [reply code/modules/recycling/sortingmachinery.dm:197]: The suggestion I made was to change it to "prime the boxcutter", you wouldn't tell someone to "open" a switchblade anyway

**Comment by Farquaar** [comment]
> You could probably do to buff its damage a bit. Unsharpened knives are at 10, cleavers and miner's knives are at 15. It's a propper tool and should be more effective than a shiv. 10 is a safe number and on par with the lesser knife tools. Should also have a wound bonus and maybe `tool_behaviour = TOOL_KNIFE`

I mean, it's a box cutter; I think the damage value seems fair for what it is. It definitely should deal less damage than a kitchen knife.

**Comment by Shadowflame909** [comment]
a damage buff to equal all other knives cool for making it useful for other situations in this sandbox. But the tool_behavior knife property would be good during the days I want to cut ian into giblets with the boxcutter instead of a knife.

It opens up options!

File: code/modules/jobs/job_types/cargo_technician.dm [hunk 10]
```diff
@@ -39,6 +39,9 @@
 	name = "Cargo Technician"
 	jobtype = /datum/job/cargo_technician
 
+	backpack_contents = list(
+		/obj/item/boxcutter = 1,
+		)
```
Comment (Mothblocks) [inline code/modules/jobs/job_types/cargo_technician.dm:44]:
```suggestion
	)
```

I know they're all like this but they're all wrong

File: code/modules/recycling/sortingmachinery.dm [hunk 11]
```diff
@@ -12,9 +12,9 @@
 /**
  * Initial check if manually unwrapping
  */
-/obj/item/delivery/proc/attempt_pre_unwrap_contents(mob/user)
+/obj/item/delivery/proc/attempt_pre_unwrap_contents(mob/user, time = 15)
```
Comment (Mothblocks) [inline code/modules/recycling/sortingmachinery.dm:15]:
```suggestion
/obj/item/delivery/proc/attempt_pre_unwrap_contents(mob/user, time = 1.5 SECONDS)
```

File: code/modules/recycling/sortingmachinery.dm [hunk 12]
```diff
@@ -189,13 +189,15 @@
...
 
 	else if(istype(item, /obj/item/boxcutter))
-		if( (item.icon_state == "boxcutter_on"))
-			unwrap_contents(user)
+		var/obj/item/boxcutter/boxcutter_item = item
+		if(boxcutter_item.on)
+			if(!attempt_pre_unwrap_contents(user, time = 5))
+				return
+			unwrap_contents()
+			to_chat(user, span_notice("You start to cut open package..."))
```
Comment (Mothblocks) [inline code/modules/recycling/sortingmachinery.dm:197]:
```suggestion
			balloon_alert(user, "cutting open package...")
```

File: code/modules/recycling/sortingmachinery.dm [hunk 13]
```diff
@@ -189,13 +189,15 @@
 		update_appearance()
 
 	else if(istype(item, /obj/item/boxcutter))
-		if( (item.icon_state == "boxcutter_on"))
-			unwrap_contents(user)
+		var/obj/item/boxcutter/boxcutter_item = item
+		if(boxcutter_item.on)
+			if(!attempt_pre_unwrap_contents(user, time = 5))
```
Comment (Mothblocks) [inline code/modules/recycling/sortingmachinery.dm:194]:
```suggestion
			if(!attempt_pre_unwrap_contents(user, time = 0.5 SECONDS))
```

File: code/game/objects/items/boxcutter.dm [hunk 14]
```diff
@@ -13,19 +13,21 @@
 	resistance_flags = FIRE_PROOF
 	force = 0
 	var/start_extended = FALSE
+	var/on = FALSE
```
Comment (Mothblocks) [inline code/game/objects/items/boxcutter.dm:16]:
```suggestion
	/// Whether or not the boxcutter has been readied
	var/on = FALSE
```

**Review by Shadyyy66 — COMMENTED** [review]
just some formatting changes

File: code/modules/recycling/sortingmachinery.dm (hunk 13 above)
Comment (Shadyyy66) [reply code/modules/recycling/sortingmachinery.dm:194]: i dont know if i want to change this to seconds, is there even any code that uses it? im worried its gonna break

File: code/game/objects/items/boxcutter.dm (hunk 3 above)
Comment (Mothblocks) [reply code/game/objects/items/boxcutter.dm:47]: ill take your word for it

File: code/modules/recycling/sortingmachinery.dm (hunk 13 above)
Comment (itseasytosee) [reply code/modules/recycling/sortingmachinery.dm:194]: Don't worry man, seconds is literally just defined as `* 10` its just for readability.
